services/statestorage/merkle/merkle_forest.go

```diff
@@ -0,0 +1,236 @@
+package merkle
+
+import (
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/crypto/hash"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+type RootId uint64
+type Proof []*Node
+
+var zeroValueHash = hash.CalcSha256([]byte{})
+
+type Node struct {
+	path     string
+	value    primitives.Sha256
+	branches [256]primitives.MerkleSha256
+}
+
+var emptyNode = &Node{value: zeroValueHash}
+var emptyNodeHash = emptyNode.hash()
+
+func createNode(path string, valueHash primitives.Sha256) *Node {
+	return &Node{
+		path:     path,
+		value:    valueHash,
+		branches: [256]primitives.MerkleSha256{},
+	}
+}
+func (n *Node) hasValue() bool {
+	return !zeroValueHash.Equal(n.value)
+}
+func (n *Node) hash() primitives.MerkleSha256 {
+	serializedNode := fmt.Sprintf("%+v", n)
```

Comment:
Recommend not to use `Sprintf` in production as a serialization mechanism, its behavior is not guaranteed to remain constant across Go versions. We should implement simple serialization ourselves

Reply:
Agreed. Final serialization is pending updates to the spec. 
Adding a //TODO to reflect this